pyomo/solvers/plugins/solvers/gurobi_direct.py

```diff
@@ -148,13 +148,31 @@ def _apply_solver(self):
                     raise
                 self._solver_model.setParam(key, float(option))
 
+        binary_vars = ComponentSet()
+        integer_vars = ComponentSet()
+        if 'relax_integrality' in self.options:
+            if self.options['relax_integrality']:
```

Comment:
I don't really like that we handle this kind of functionality through the solver options dictionary. That seems like it should be used for solver specific options only. Could this be implemented through a keyword to the solve method? Again though, I don't want to prevent you from getting other work done, so I will approve this PR either way.

Reply:
I completely agree. I only did it this way to match the behavior of the lp interface... If we want another solution, such as the one you suggested, that is perfectly fine with me.